aerogear/unifiedpush-operator
# Update grafana-operator dependency

## Motivation
JIRA: https://issues.redhat.com/browse/INTLY-10785

## What
Grafana-operator dependency was updated in order to pickup newer struct definitions for the CRs provided by it.

## Why
In order to avoid Unmarshalling errors.

## How
Basically `dep ensure -update github.com/integr8ly/grafana-operator`
and minor code changes to remove extra field.

## Verification Steps
Have the previous version of the operator running.
Verify that an error like this is not present in operator logs:
```
E0819 20:43:53.556597   1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.GrafanaDashboard: v1alpha1.GrafanaDashboardList.Items: []v1alpha1.GrafanaDashboard: v1alpha1.GrafanaDashboard.Status: v1alpha1.GrafanaDashboardStatus.Phase: readUint64: unexpected character: ÿ, error found in #10 byte of ...|,"phase":"reconcilin|..., bigger context ...|54dd6c3f2f8","id":15,"message":"success","phase":"reconciling","slug":"unifiedpush-server","uid":"Zc|...
```
Start an API proxy `oc proxy --port 1234`
Update the namespace part("redhat-rhmi-ups") in the command below and execute it:
```
curl -s localhost:1234/apis/integreatly.org/v1alpha1/namespaces/redhat-rhmi-ups/grafanadashboards/unifiedpushserver-dashboard/status -XPATCH  -H "Accept: application/json" -H "Content-Type: application/merge-patch+json" -d '{"status": {"hash": "3de09b7d992c2cdac355954dd6c3f2f8", "id": 15, "message": "success", "phase": "reconciling", "slug": "unifiedpush-server", "uid": "ZckQWaPMk"}}'
```
This will add legacy status fields to the GrafanaDashboard CR.
Restart the operator.
Now you should see error mentioned above in the operator logs.
Stop your instance of the operator.
Run the operator from this branch.
You shouldn't see the error in the logs, instead you should see a message indicating successful reconcile.

## Checklist:

## Progress
- [ ] Run the release pipeline once this is merged
